src/utils/errorHelpers.tsx

```diff
@@ -0,0 +1,17 @@
+import type { ExpirationDateError, CardNumbersError } from "../types";
+
+// 유효기간 검증에서 가장 첫번째 에러 메시지 반환
+export const getFirstExpirationErrorMessage = (
+  error: ExpirationDateError
+): string => {
+  if (error.month.hasError) return error.month.errorMessage;
+  if (error.year.hasError) return error.year.errorMessage;
+
+  return "";
+};
+
+// 카드번호 검증에서 가장 첫번째 에러 메시지 반환
+export const getFirstErrorMessage = (error: CardNumbersError): string => {
+  const errorBlock = Object.values(error).find((block) => block.hasError);
+  return errorBlock ? errorBlock.errorMessage : "";
+};
```

Comment:
일관된 방법으로 에러 우선 순위를 결정하는 건 어떨까요?

```suggestion
const ERROR_PRIORITY = {
  EXPIRATION_DATE: ["month", "year"] as const,
  CARD_NUMBER: [
    "firstBlock",
    "secondBlock",
    "thirdBlock",
    "fourthBlock",
  ] as const,
} as const;

export const getFirstExpirationErrorMessage = (
  error: ExpirationDateError
): string => {
  for (const field of ERROR_PRIORITY.EXPIRATION_DATE) {
    if (error[field].hasError) return error[field].errorMessage;
  }
  return "";
};

export const getFirstErrorMessage = (error: CardNumbersError): string => {
  for (const field of ERROR_PRIORITY.CARD_NUMBER) {
    if (error[field].hasError) return error[field].errorMessage;
  }
  return "";
};
```

Reply:
```
// constants/cardConstants.ts
export const CARD_INFORMATION = {
  EXPIRATION_DATE: ["month", "year"] as const,
  CARD_NUMBER_BLOCK: [
    "firstBlock",
    "secondBlock",
    "thirdBlock",
    "fourthBlock",
  ] as const,
};
```
- utils/cardNumberValidators.tsx의 validCardNumbersBlock에서도 keys로  CARD_NUMBER_BLOCK 부분을 사용해야 해서, 말씀주신 것처럼 아예 배열을 constants 안에 넣었습니다! 이렇게 배열을 상수화로 사용하는 방법도 앞으로 유용하게 사용할 수 있을 것 같네요! 

```
// utils/cardNumberValidators.tsx
// 카드 번호 블록 검증
export const validCardNumbersBlock = (
  cardNumbers: CardData["numbers"]
): CardFormError["numbers"] => {
(생략)

  CARD_INFORMATION.CARD_NUMBER_BLOCK.forEach((key, index) => {
    if (index === 0) return;

    result[key] = validBlockIsFull(
      cardNumbers[CARD_INFORMATION.CARD_NUMBER_BLOCK[index - 1]],
      cardNumbers[CARD_INFORMATION.CARD_NUMBER_BLOCK[index]]
    );
  });

  return result;
};


// utils/errorHelpers.tsx
// 유효기간 검증에서 가장 첫번째 에러 메시지 반환
export const getFirstExpirationErrorMessage = (
  error: CardFormError["expirationDate"]
): string => {
  for (const field of CARD_INFORMATION.EXPIRATION_DATE) {
    if (error[field].hasError) return error[field].errorMessage;
  }

  return EMPTY_STRING;
};
```